spec/unleash/metrics_reporter_spec.rb

```diff
@@ -112,6 +113,35 @@
     expect(WebMock).to_not have_requested(:post, 'http://test-url/client/metrics')
   end
 
+  it "generates an empty report when no metrics after 10 minutes" do
+    WebMock.stub_request(:post, "http://test-url/client/metrics")
+      .to_return(status: 200, body: "", headers: {})
+    Unleash.configure do |config|
+      config.url      = 'http://test-url/'
+      config.app_name = 'my-test-app'
+      config.instance_id = 'rspec/test'
+      config.disable_client = true
+    end
+    Unleash.engine = YggdrasilEngine.new
+
+    metrics_reporter.last_time = Time.now - 601
+    report = metrics_reporter.generate_report
+    expect(report[:bucket]).to be_empty
+
+    metrics_reporter.post
+
+    expect(WebMock).to have_requested(:post, 'http://test-url/client/metrics')
+      .with(
+        body: hash_including(
+          yggdrasilVersion: anything,
+          specVersion: anything,
+          platformName: anything,
+          platformVersion: anything,
+          bucket: {}
+        )
+      )
```

Comment:
Is this working as expected? I thought the intention would be _not_ to post to server when it's empty.

Reply:
Nvm, we're subtracting by 10 minutes and 1 second, which means that the second part of our condition is not met, so we post the empty metrics. Got it 👍

Reply:
Indeed, turns out this code path was apparently not tested and didn't work as expected. Although it's an edge case I think it makes sense